framework/doc/content/documentation/application_development/performance_benchmarking.md

```diff
@@ -1,39 +1,93 @@
 
-benchmark.py
--------------------
+Performance Benchmarking
+=========================
 
-This script provides functionality for benchmarking and tracking MOOSE performance.  It can be
-used to run benchmarks, generate trend visualizations, and look at stats comparing bencharks
-between various revisions.  Normally, benchmarks should be run through the test harness (i.e.
-using the ``run_tests`` script) by doing e.g. ``./run_tests --run speedtests``
+Utilities for doing performance benchmarking of MOOSE-based applications are included in the main
+MOOSE repository.  These utilities provide functionality for benchmarking and tracking MOOSE
+performance.  They can be used to run benchmarks, generate trend visualizations, and look at stats
+comparing bencharks between various revisions.  The following sections describe how to setup a
+benchmark machine and use it to run benchmarks and visualize results.
+
+Tuning a Benchmarking Machine
+------------------------------
 
 In order to obtain accurate results, you need to run the benchmark process(es)
 as close to isolated as possible.  On a linux system, you should e.g. use cpu
 isolation via setting kernel boot parameters:
 
- ```
+```
```

Comment:
What is the language? bash, etc. Use "text" if you don't know.

Reply:
You should put a language on all of the code blocks if possible, it makes the syntax highlighting better.

Reply:
text